### PR TITLE
Add ability to create and edit flappy levels in levelbuilder

### PR DESCRIPTION
**Pull request by Hamms**

### CODE REVIEW

File: dashboard/app/models/game.rb [hunk 1]
```diff
@@ -51,6 +51,10 @@ def self.custom_maze
   LEVEL_GROUP = 'level_group'.freeze
   PUBLIC_KEY_CRYPTOGRAPHY = 'public_key_cryptography'.freeze
 
+  def self.flappy
+    @@game_flappy ||= find_by_name("Flappy")
```
Comment (joshlory) [inline dashboard/app/models/game.rb]: One minor thing to note: the `Game` entry for flappy includes an intro video.

```
Flappy:flappy:flappy_intro
```

File: dashboard/app/models/levels/flappy.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  end
+
+  # List of possible skins, the first is used as a default.
+  def self.skins
+    %w(flappy)
+  end
+
+  def self.create_from_level_builder(params, level_params)
+    puts params
+    puts level_params
```
Comment (joshlory) [inline dashboard/app/models/levels/flappy.rb]: Extra logging.

File: dashboard/app/models/levels/flappy.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+class Flappy < Blockly
+  serialized_attrs %w(
+    success_condition
+    failure_condition
+    default_flap
+    free_play
+    ground
+    obstacles
+    score
+    shareable
```
Comment (joshlory) [inline dashboard/app/models/levels/flappy.rb:35]: For other puzzle types, is this determined by the `free_play` flag?

Comment (Hamms) [reply dashboard/app/models/levels/flappy.rb:35]: It's unfortunately pretty inconsistent; all level types consider the free_play flag when deciding whether or not to share, but they all have their own extra bits of logic, too. For flappy, sharing is determined by `level.freePlay && level.shareable`.

File: dashboard/app/models/game.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/app/models/game.rb]: It's not clear to me what about that I should be noting

Comment (joshlory) [reply dashboard/app/models/game.rb]: Ah, sorry!  It means that any Flappy level will always have `flappy_intro` as an autoplay video.  Via [LevelsHelper#select_and_track_autoplay_video](https://github.com/code-dot-org/code-dot-org/blob/07fdda15d346db7a4b091b8f3ec00531dac39406/dashboard/app/helpers/levels_helper.rb#L84) -> [Level#related_videos](https://github.com/code-dot-org/code-dot-org/blob/1fd7c7208d78c037860d3ea3b92b810a7d4546ea/dashboard/app/models/levels/level.rb#L90).

```ruby
def related_videos
  ([game.intro_video, specified_autoplay_video] + concepts.map(&:video)).compact.uniq
end

autoplay_video = @level.related_videos.find {|video| !client_state.video_seen?(video.key)}
```

Comment (Hamms) [reply dashboard/app/models/game.rb]: Given that the same is true of Maze, Artist, and Farmer, I think that's fine

Comment (joshlory) [reply dashboard/app/models/game.rb]: I believe maze levels use the `CustomMaze` game, and artist levels use the `Custom` game.

```
Custom:turtle
CustomMaze:maze
```

Comment (Hamms) [reply dashboard/app/models/game.rb]: Well, damn.

Comment (joshlory) [reply dashboard/app/models/game.rb]: I'm actually not sure which behavior we want 😛 but wanted note that this was a side effect.

**Pull request merged**